### DISCUSSION TITLE
Saving snapshot as YAML or JSON

### DISCUSSION THREAD

**Discussion by cmordi** [microsoft/Qcodes#5639] [General]
Hi, 

Is there a current method to save the current state of my parameters and values that are shown in the `print_readable_snapshot` call and then make it able to load later? I've looked into databases but that seems like too much since i'm not really taking measurements and I only have one instrument.

![image](https://github.com/microsoft/Qcodes/assets/19673888/e4dc1ddd-3008-4e43-9ea1-fa3943c69cb8)

This is what I get from the `print_readable_snapshot`. I simply want the ability to save the current state of those parameters and values, preferably as a JSON or YAML file and also the ability to load them in later as an experiment.

**Comment by jenshnielsen** [comment]
`print_readable_snapshot` is meant as a wrapper around the `snapshot` method of the instrument. We dont provide a method to directly save the snapshot but this snapshot is in json compatible format so you should be able to do something like

```python
import json
from qcodes.utils import NumpyJSONEncoder

snapshot = myinstr.snapshot()

with open("pathtofileiwanttowriteto")" as f
    json.dump(snapshot, cls=NumpyJSONEncoder)
```

The NumpyJSONEncoder is required to enable us to write numpy arrays to the json file as regular arrays

**Reply by cmordi** [reply]
Thanks for the idea Jens. Is there a possible way to load this data? For example if I have it saved in JSON format can I load but not overwrite an experiment with it? I believe I read over a way to load a YAML file but that involved load in the station configurations.

**Reply by jenshnielsen** [reply]
We don't have support for setting the state of an instrument from the snapshot at the moment thou I can see that being a potentially useful feature

**Reply by cmordi** [reply]
No problem, I could also see it being useful. Thanks for your assistance!

**Reply by astafan8** [reply]
@cmordi you're welcome to submit a PR implementing a first version of the loading-from-snapshot feature :)

**Reply by cmordi** [reply]
@astafan8 I'll try my best to implement it ;)
